yt_dlp/extractor/nzonscreen.py

```diff
@@ -45,49 +49,109 @@ class NZOnScreenIE(InfoExtractor):
-                'height': int_or_none(playlist.get('height')) if id_ == 'hi' else None,
-                'width': int_or_none(playlist.get('width')) if id_ == 'hi' else None,
-                'filesize_approx': float_or_none(traverse_obj(playlist, ('h264', f'{id_}_res_mb')), invscale=1024**2),
-            }
+                playlist, ('h264', {'lo': 'lo_res', 'hi': 'hi_res', 'hd': 'hd_res'}),
+                expected_type=url_or_none).items()):
+            if traverse_obj(playlist, ('h264', f'{id_}_res_mb', {float_or_none})):
+                formats.append({
+                    'url': url,
+                    'format_id': id_,
+                    'ext': 'mp4',
+                    'quality': quality,
+                    'filesize_approx': float_or_none(traverse_obj(
+                        playlist, ('h264', f'{id_}_res_mb')), invscale=1024**2),
+                })
+        if formats:
+            formats[-1].update(traverse_obj(playlist, {
+                'height': ('height', {int_or_none}),
+                'width': ('width', {int_or_none}),
+            }))
+        return formats
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
+        title = strip_or_none((
+            self._html_extract_title(webpage, default=None)
+            or self._og_search_title(webpage)).rsplit('|', 2)[0])
+        playlist = self._download_json(
+            f'https://www.nzonscreen.com/html5/video_data/{video_id}', video_id,
+            'Downloading media data')
 
-        playlist = self._parse_json(self._html_search_regex(
-            r'data-video-config=\'([^\']+)\'', webpage, 'media data'), video_id)
+        if not self._yes_playlist(video_id, video_id):
+            del playlist[1:]
 
-        return {
-            'id': playlist['uuid'],
+        return self.playlist_result([{
+            'alt_title': title if len(playlist) == 1 else None,
             'display_id': video_id,
-            'title': strip_or_none(playlist.get('label')),
-            'description': strip_or_none(playlist.get('description')),
-            'alt_title': strip_or_none(remove_end(
-                self._html_extract_title(webpage, default=None) or self._og_search_title(webpage),
-                ' | NZ On Screen')),
-            'thumbnail': traverse_obj(playlist, ('thumbnail', 'path')),
-            'duration': float_or_none(playlist.get('duration')),
-            'formats': list(self._extract_formats(playlist)),
             'http_headers': {
                 'Referer': 'https://www.nzonscreen.com/',
                 'Origin': 'https://www.nzonscreen.com/',
             },
-        }
+            'subtitles': {'en': [{
+                'url': traverse_obj(playinfo, ('h264', 'caption_url', {urljoin('https://www.nzonscreen.com')})),
+                'ext': 'vtt',
+            }]},
+            **traverse_obj(playinfo, {
+                'formats': {self._extract_formats},
```

Comment:
We don't want errors to be suppressed here, correct?

```suggestion
            'formats': self._extract_formats(playinfo),
            **traverse_obj(playinfo, {
```